PhysicsTools/PatUtils/src/MiniIsolation.cc

```diff
@@ -49,4 +55,22 @@ PFIsolation getMiniPFIsolation(const pat::PackedCandidateCollection *pfcands,
 
 }
 
+  float relMiniIsoPUCorrected(const PFIsolation& iso,
+			      const math::XYZTLorentzVector& p4,
+			      float dr,
+			      float rho)
+  {
+    double absEta = fabs(p4.eta());
+    double ea = 0;
+    //Spring15 version
+    if      (absEta<0.800) ea = 0.0735;
```

Comment:
rumor has it that this is actually buggy for the current version of fixedGridRhoFastjetCentralNeutral.
The constants were derived on 74X "fixedGridRhoFastjetCentralNeutral", which actually had a very different content.

Reply:
That's what we have at the moment. As soon as a better option is available we will update.

Reply:
Where in this code does it say that it is a muon?
Constants are different for muons and electrons
https://twiki.cern.ch/twiki/bin/view/CMS/SUSLeptonSF#ID_IP_ISO

Also, since values can change for different data taking periods, these better come from config parameters.

Reply:
This is not resolved yet.
Perhaps rename to ```muonRelMiniIsoPUCorrected ``` ?

Reply:
Not sure I follow. Is there any other implementation? I think it's better to leave it as is and change it when either we have a new approach or e-gamma wants to change it as well. So far I don't see anything muon specific here.

Reply:
Missed that. The fix is coming.

Reply:
Dedicated studies (see [this](https://indico.cern.ch/event/666431/#3-inconsistency-in-effectivear) talk) show that using _fixedGridRhoFastjetAll_ reduces pileup dependence.

@drkovalskyi probably it's worth updating this now?
The effective areas will have to be rederived on 9_4_X samples in any case, but I think that change will be minor with respect to fixing the rho definition.

We are going to update the NanoAOD code to use the muon information added by this PR.
The current implementation of pileup subtraction in the cms-nanoAOD repository (not yet merged in CMSSW) has already switched to _fixedGridRhoFastjetAll_ with unchanged EAs.

@gpetruc @arizzi @emanueledimarco @gzevi

Reply:
Thanks Marco. Will follow up.